lib/optional-keys/index.ts

```diff
@@ -1,5 +1,5 @@
-export type OptionalKeys<Type> = Type extends unknown
-  ? {
-      [Key in keyof Type]-?: undefined extends { [Key2 in keyof Type]: Key2 }[Key] ? Key : never;
-    }[keyof Type]
+export type OptionalKeys<Type> = Type extends object
+  ? keyof {
+      [Key in keyof Type as Type extends Required<Pick<Type, Key>> ? never : Key]: any;
```

Comment:
(Optional) Anything can be a value, you can simply use a numeric literal as the shortest option (e.g. 1) to avoid usage of `any`

Reply:
IMO, it's better to have `any` just to signify that the value can be anything. Having something more specific, like `1`, might not convey the same meaning and could create confusion.

Can change it to `unknown` though, if `any`s are to be avoided.

Reply:
There are no best practices in this repo (yet), so `any` works as well. I'll leave it to you, feel free to resolve the comment if you'd like to stay with `any`

Reply:
Let's leave it as `any` then.